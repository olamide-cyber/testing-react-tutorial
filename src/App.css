.app,
.decrement {
  margin-left: 20px;
}

.increment,
.decrement {
  padding: 8px;
}

.decrement:disabled {
  cursor: not-allowed;
}

.email {
  margin-left: 18px;
  display: inline-block;
  padding: 5px;
  font-weight: bold;
  color: #282c34;
  cursor: pointer;
}

.email:hover {
  text-decoration: underline;
  color: #637db1;
  transition: all .2s ease-in-out;
}








/* 
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
 */