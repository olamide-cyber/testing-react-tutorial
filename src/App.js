import React, { useState } from 'react';
import './App.css';
import Users from './Users/Users';
import MyComponent from './__mocks__/MyComponent';

function App () {
    const [count, setCount] = useState(0);

    const makeIncrementer = amount => setCount(prev => prev + amount);

    const increment = () => makeIncrementer(1);

    const decrement = () => {
      if (count === 0) {
        return;
      }
      return makeIncrementer(-1);
    }

    return (
        <div className="app">
            <p>Count: {count}</p>
            <button 
                className="increment" 
                onClick={increment}
            >
                Increment
            </button>
            <button 
                disabled={count === 0 ? true : false} 
                className="decrement" 
                onClick={decrement}
            >
                Decrement
            </button>
            <div className="users">
                <Users />
                <MyComponent />
            </div>
        </div>
    );
}

export default App;
