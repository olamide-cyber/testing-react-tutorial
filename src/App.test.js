import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import { render } from '@testing-library/react';

const MockMyComponent = () => {
  React.useEffect(() => {
    console.log('using an effect');
  });
  return (<div>Hello World</div>);
};
jest.mock('./MyComponent', () => MockMyComponent);


describe('MyComponent', () => {
  it('renders', () => {
    const { container } = render(<App/>);
    expect(container.textContent).toMatch('Hello World');
  })
});

describe('App component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('p').text();
    expect(text).toBe('Count: 0')
  })

  it('incremenent by one when the increment button is clicked', () => {
    const wrapper = shallow(<App />);
    const incrementBtn = wrapper.find('button.increment');
    incrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 1');
  });

  it('decrement count when the decrement button is clicked', () => {
    const wrapper = shallow(<App />);
    const decrementBtn = wrapper.find('button.decrement');
    decrementBtn.simulate('click');
    const text = wrapper.find('p').text();
    expect(text).toEqual('Count: 0');
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});







// import { render, screen } from '@testing-library/react';
// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });