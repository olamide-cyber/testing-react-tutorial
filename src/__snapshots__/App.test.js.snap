// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="app"
>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="decrement"
    disabled={true}
    onClick={[Function]}
  >
    Decrement
  </button>
  <div
    className="users"
  >
    <h1>
      Loading usernames, please wait...
    </h1>
    <div>
      Hello World
    </div>
  </div>
</div>
`;
